src/Symfony/Component/Console/Command/CommandConfiguration.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Input\InputDefinition;
+
+/**
+ * Configuration of a command.
+ *
+ * @author Nikita Konstantinov
+ *
+ * @api
+ */
+final class CommandConfiguration
+{
+    private $name;
+    private $aliases = array();
+    private $definition;
+    private $enabled;
+    private $processTitle;
+    private $help;
+    private $description;
+    private $synopsis;
+
+    /**
+     * @param string $name
+     * @return CommandConfiguration
+     */
+    public static function create($name)
+    {
+        return new self($name);
+    }
+
+    /**
+     * @param string $name
+     * @param InputDefinition $definition
+     * @param string[] $aliases
+     * @param bool $enabled
+     * @param string $processTitle
+     * @param string $help
+     * @param string $description
+     * @param string $synopsis
+     */
+    public function __construct(
+        $name = null,
+        InputDefinition $definition = null,
+        array $aliases = array(),
+        $enabled = true,
+        $processTitle = null,
+        $help = null,
+        $description = null,
+        $synopsis = null
+    ) {
+        if ($name !== null) {
+            $this->validateName($name);
+        }
+
+        foreach ($aliases as $alias) {
+            $this->validateName($alias);
+        }
+
+        $this->name = $name;
+        $this->aliases = $aliases;
+        $this->definition = $definition ?: new InputDefinition();
+        $this->enabled = $enabled;
+        $this->processTitle = $processTitle;
+        $this->help = $help;
+        $this->description = $description;
+        $this->synopsis = $synopsis;
+    }
+
+    /**
+     * Whether the command is enabled or not in the current environment.
+     *
+     * @return bool
+     */
+    public function isEnabled()
+    {
+        return $this->enabled;
+    }
+
+    /**
+     * @param InputDefinition $definition
+     * @return CommandConfiguration
+     */
+    public function withDefinition(InputDefinition $definition)
```

Comment:
your design for this class looks weird: it looks like you are trying to make it immutable, but you embed a mutable InputDefinition inside it, meaning that it is **not** immutable